src/andromede/simulation/optimization.py

```diff
@@ -579,161 +678,192 @@ def make_constraint(
     return solver_constraints
 
 
-def _create_variables(
-    network: Network,
-    opt_context: OptimizationContext,
-    solver: lp.Solver,
-) -> None:
-    for component in network.all_components:
-        component_context = opt_context.get_component_context(component)
-        model = component.model
+class OptimizationProblem:
+    class Type(Enum):
+        """
+        Class to specify the type of the created problem:
+            - master: Creates a Xpansion master problem with investment and coupling variables and constraints only
+            - subproblem: Creates Xpansion sub-problems with operational and coupling variables and constraints only
+            - merged: Creates a Antares Simulator/ Xpansion problem with all
+        """
 
-        for model_var in model.variables.values():
-            var_indexing = IndexingStructure(
-                model_var.structure.time, model_var.structure.scenario
-            )
-            instantiated_lb_expr = None
-            instantiated_ub_expr = None
-            if model_var.lower_bound:
-                instantiated_lb_expr = _instantiate_model_expression(
-                    model_var.lower_bound, component.id, opt_context
+        merged = 0
+        master = 1
+        subproblem = 2
+
+    name: str
```

Comment:
From what I knew, if you only annotate the type, the interpreter doesn't consider it as a class variable.
It may have changed since, but this post [here](https://stackoverflow.com/questions/44959876/type-hints-convention-for-instance-variables-python/69115321#69115321) seems to agree with me.

Reply:
I created a small [snippet](https://www.online-python.com/4wQWzDVJMI) to check and it seems to work as I intended

Reply:
Indeed thanks for sharing the knowledge :)